### PR TITLE
Rename `ctranspose` to `adjoint`

### PR DESCRIPTION
**Pull request by andyferris**
Just to prove exactly how seriously we take transposes, we need to rename one of them.

Part of JuliaLang/LinearAlgebra.jl#408. This one was simple find-replace (so far).

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -309,6 +309,9 @@ Deprecated or removed
   * `Base.cpad` has been removed; use an appropriate combination of `rpad` and `lpad`
     instead ([#23187]).
 
+  * `ctranspose` and `ctranspose!` have been deprecated in favor of `adjoint` and `adjoint!`,
+    respectively.
```
Comment (Sacha0) [inline NEWS.md]: Add the pull request number? :)

File: base/deprecated.jl [hunk 2]
```diff
@@ -1684,6 +1684,10 @@ end
 # PR #22742: change in isapprox semantics
 @deprecate rtoldefault(x,y) rtoldefault(x,y,0) false
 
+# PR #23235
+@deprecate ctranspose adjoint
+@deprecate ctranspose! adjoint
```
Comment (Sacha0) [inline base/deprecated.jl]: Missing `!`?

File: test/operators.jl [hunk 3]
```diff
@@ -49,7 +49,7 @@ p = 1=>:foo
 @test xor(2) == 2
 @test (⊻)(2) == 2
 
-# @test ctranspose('a') == 'a' # (c)transpose of Chars no longer supported
+# @test adjoint('a') == 'a' # (c)transpose of Chars no longer supported
```
Comment (Sacha0) [inline test/operators.jl]: Perhaps update the comment as well?

**Review by StefanKarpinski — APPROVED** [review]

File: test/operators.jl (hunk 3 above)
Comment (andyferris) [inline test/operators.jl]: Hmm... I think we can delete this now.

(I have an upcoming PR which will let us transpose arrays of `Char`s and `String`s and so-on, so the context of this whole line will be rather out-of-date.)

**Comment by andyferris** [comment]
Any more comments or any detractors before I merge this?

(I just realized we should probably rename `A_mul_Bc` to `A_mul_Ba` etc, but I *won't* do that because we plan to remove these entirely in this release cycle.)

**Comment by StefanKarpinski** [comment]
Let's give it a bit more time since this has only been open for half a day on a weekend. I don't imagine there being much controversy here, however.

**Comment by andyferris** [comment]
Something went funky with the linux tests

**Comment by StefanKarpinski** [comment]
Restarted them.

**Comment by StefanKarpinski** [comment]
Bump – rebase. That Linux issue should be resolved now.

**Comment by ViralBShah** [comment]
cc @alanedelman

**Comment by andyferris** [comment]
Thanks @andreasnoack (I'm travelling ATM). 

The test errors appear unrelated and oddly spurious... one inference failure for RowVector and test failure one in codegen, but only on specific builds.

**Pull request merged**